### ISSUE TITLE
Get all documents feature

### ISSUE THREAD

**Issue by DRK3** [decentralized-identity/secure-data-store#111]
Hi!

It would be useful if data vaults have some sort of a "Get all documents within a given vault" feature, along with some sort of pagination syntax. Are there any plans to add such a feature, or could it be considered?

Thanks!

**Comment by DRK3** [comment]
Use case: being able to retrieve all documents without having to store the location/ID of every document somewhere else.

**Comment by agropper** [comment]
> It would be useful if data vaults have some sort of a "Get all documents within a given vault" feature, along with some sort of pagination syntax. Are there any plans to add such a feature, or could it be considered?

This is pretty much my point in calling for structured scopes: https://github.com/decentralized-identity/secure-data-store/issues/107#issuecomment-700395500

**Comment by dmitrizagidulin** [comment]
@DRK3 - I agree, this would be a useful feature.

Incidentally, a pagination syntax would also help us in searches on an encrypted index... 

So the way I see it, this raises a couple of questions:

1. Should we add a separate API call (something like `GET /vaults/<vault id>/documents`)? The other alternative would be to define a default/virtual Index on `_vault` or something. So that "give me all the documents in a vault" becomes "search for all of the documents that match this `_vault` Index value".
2. What syntax, exactly, should we use for pagination? There's a handful of semi-standards for this. We could look at:
    * [JSON-API Pagination](https://jsonapi.org/examples/#pagination)
    * [Elastic Search Pagination](https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html)
    * MongoDB's sort / skip / limit syntax?
    * [Microsoft's API Guidelines for Pagination](https://github.com/Microsoft/api-guidelines/blob/vNext/Guidelines.md#98-pagination)?
    * Other database pagination syntax? CouchDB, Postgres, etc...

**Comment by agropper** [comment]
Indexes are fine but access would be subject to authorization just like the
documents themselves which brings us to structured scopes again.

- Adrian

**Comment by dmitrizagidulin** [comment]
@agropper of course, fetching the list of documents in a vault (either by a direct API or an index search) would certainly be subject to authorization.

**Comment by DRK3** [comment]
Something I had only realized later: you could use the indexing feature to achieve this behaviour by just having everything in a vault belong to some sort of index that you can query later. Not sure if there's any value in my original request anymore.
